packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart

```diff
@@ -79,21 +77,16 @@ abstract class CameraPlatform extends PlatformInterface {
     throw UnimplementedError('prepareForVideoRecording() is not implemented.');
   }
 
-  /// Starts a video recording.
-  ///
-  /// If no [file] parameter is provided, the recording will be saved to a new
-  /// file in a temporary location.
+  /// Starts a video recording and returns the file where it will be saved.
   ///
   /// The file is written on the flight as the video is being recorded.
-  /// If a file already exists at the path for the provided file instance,
-  /// an error will be thrown.
   /// The file can be read as soon as [stopVideoRecording] returns it.
-  Future<XFile> startVideoRecording(int cameraId, {XFile file}) {
+  Future<XFile> startVideoRecording(int cameraId) {
     throw UnimplementedError('startVideoRecording() is not implemented.');
   }
 
-  /// Stops the video recording and returns the file where it was saved.
-  Future<XFile> stopVideoRecording(int cameraId) {
+  /// Stops the video recording.
```

Comment:
nit: would be nice to mention again that when the `stopVideoRecording` method completes the `XFile` returned from `startVideoRecording` can be used. Something along these lines:
```suggestion  /// Stops the video recording.
  ///
  /// When the [stopVideoRecording] method completes successfully the recorded
  /// video can be accessed through the file returned by the 
  /// [startVideoRecording] method.
```